packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -707,6 +695,46 @@ class ZoomPageTransitionsBuilder extends PageTransitionsBuilder {
   // for the Impeller backend.
   static const bool _kProfileForceDisableSnapshotting = bool.fromEnvironment('flutter.benchmarks.force_disable_snapshot');
 
+  @override
+  DelegatedTransition? get delegatedTransition => ZoomDelegatedTransition(
+    builder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child) => snapshotAwareDelegatedTransition(context, animation, secondaryAnimation, child, allowSnapshotting, allowEnterRouteSnapshotting),
+  );
+
+  /// A transition builder that takes in account the snapshotting properties of
+  /// [ZoomPageTransitionsBuilder].
+  ///
+  /// This is provided to [ZoomDelegatedTransition.builder], to be passed to the
+  /// previous page route, when the current page is added to the stack.
+  static Widget snapshotAwareDelegatedTransition(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child, bool allowSnapshotting, bool allowEnterRouteSnapshotting) {
```

Comment:
Are you sure this should be public?

Reply:
This is used elsewhere in the file in `_ZoomPageTransition` to reduce code duplication. If it was made private it would have to be passed in a different way.